packages/plugin-core/src/commandFactory.ts

```diff
@@ -0,0 +1,37 @@
+/**
+ * Class responsible for creation of commands, specifically useful when one
+ * command needs an instance of another command, this adds in a layer of
+ * decoupling.
+ */
+import { BaseCommand } from "./commands/base";
+import {
+  ShowPreviewCommandOpts,
+  ShowPreviewCommandOutput,
+} from "./commands/ShowPreviewInterface";
+import { ShowPreviewCommand } from "./commands/ShowPreview";
+import vscode from "vscode";
+import { ICommandFactory } from "./commandFactoryInterface";
+import {
+  GoToNoteCommandOpts,
+  GoToNoteCommandOutput,
+} from "./commands/GoToNoteInterface";
+import { GotoNoteCommand } from "./commands/GotoNote";
+import { IDendronExtension } from "./dendronExtensionInterface";
+
+export class CommandFactory implements ICommandFactory {
```

Comment:
This allows commands that depend on instances of other commands to get instances of those commands without creating tight coupling to those implementations.